### PR TITLE
Fix/too big amount

### PR DESCRIPTION
**Pull request by lpascal-ledger**
This PR trims leading `0`s on `PB_BYTES_ARRAY_T()` used to host amounts. It fixes a display error of the amounts during a SWAP, a SELL or a FUND transaction.

Currently, the exchange application accepts amount such as `\x00\x00\x00\x00\x00\x00\x00\x01\x02` (9B) and stores it as is.
When sending this amount to a coin application, it may conflict with application's constraints and fail.
For instance, both the Bitcoin applications does not accept amount bigger than 8B.
By removing the leading `0`s, this change allows left-padded amount to be correctly managed.

### CODE REVIEW

File: test/src/bugs.test.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    var tr = new proto.ledger_swap.NewTransactionResponse();
+    tr.setPayinAddress("0xd692Cb1346262F584D17B4B470954501f6715a82");
+    tr.setPayinExtraId("");
+    tr.setRefundAddress("bc1qwpgezdcy7g6khsald7cww42lva5g5dmasn6y2z");
+    tr.setRefundExtraId("");
+    tr.setPayoutAddress("bc1qwpgezdcy7g6khsald7cww42lva5g5dmasn6y2z");
+    tr.setPayoutExtraId("");
+    tr.setCurrencyFrom("BTC");
+    tr.setCurrencyTo("BTC");
+    // 1 BTC to 1 BTC
```
Comment (bigspider) [inline test/src/bugs.test.js:41]: outdated comment, amount in snapshot swaps 2.69... with 1 BTC.

File: test/src/bugs.test.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    TRANSACTION_RATES,
+    TRANSACTION_TYPES
+} from "./exchange.js";
+
+import Zemu from "@zondax/zemu";
+import { waitForAppScreen, zemu } from './test.fixture';
+
+// Forcing AmountToWallet to be bigger than 8B, so that the Bitcoin app would
+// accept it only if the trim function works as expected
+test('[Nano S] Overflow values should be trimmed when swaping', zemu("nanos", async (sim) => {
```
Comment (bigspider) [inline test/src/bugs.test.js:27]: typo, "swapping"

File: test/src/bugs.test.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    // Wait until we are not in the main menu
+    await waitForAppScreen(sim);
+    await sim.navigateAndCompareSnapshots('.', 'nanos_btc_to_btc_overload_swap', [4, 0]);
+    await expect(checkRequest).resolves.toBe(undefined);
+
+    await swap.signCoinTransaction();
+
+    await Zemu.sleep(1000);
+
+    let transport = await sim.getTransport();
```
Comment (bigspider) [inline test/src/bugs.test.js:65]: does this do anything? I think the last three lines can be removed

File: test/src/bugs.test.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    // useless what to put instead?
+    const btcAddressParams = await getSerializedAddressParametersBTC("84'/0'/0'/1/0", "bech32");
+
+    const checkAssetIn = swap.checkPayoutAddress(BTCConfig, BTCConfigSignature, btcAddressParams.addressParameters);
+
+    // Wait until we are not in the main menu
+    await waitForAppScreen(sim);
+    await sim.navigateAndCompareSnapshots('.', 'nanos_valid_funding_is_accepted', [5, 0]);
+
+    await swap.signCoinTransaction();
```
Comment (bigspider) [inline test/src/bugs.test.js:115]: as for the other test, probably this line can be removed

File: test/src/bugs.test.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    const message = Buffer.concat([Buffer.from('.'), base64_payload])
+    const digest: Buffer = Buffer.from(sha256.sha256.array(message));
+    const lol = secp256r1.sign(digest, fundTestPrivateKey).signature;
+
+    process.stdout.write(JSON.stringify(lol) + '\n');
+
+    const signature = secp256r1.signatureExport(lol)
+    //console.log("Length of generated signature %d", sigTemp.length)
+
+    process.stdout.write(JSON.stringify(signature) + '\n');
```
Comment (bigspider) [inline test/src/bugs.test.js:102]: remove debug prints, and rename `lol` to something :P

**Review by bigspider — CHANGES_REQUESTED** [review]
Looks good in general, just some nits here and there.

It could be nice to find a better name for the `bugs.test.js` file, or move those tests in some other test module.

File: test/src/bugs.test.js (hunk 5 above)
Comment (lpascal-ledger) [reply test/src/bugs.test.js:102]: lol

Comment (lpascal-ledger) [reply test/src/bugs.test.js:102]: My patience with JS illustrated right there.

**Review by bigspider — APPROVED** [review]

**Review by apaillier-ledger — APPROVED** [review]

**Pull request merged**